addons/purchase_stock/models/account_move_line.py

```diff
@@ -327,7 +327,10 @@ def _generate_price_difference_vals(self, layers):
             unit_valuation_difference_curr = unit_valuation_difference * self.currency_rate
             unit_valuation_difference_curr = product_uom._compute_price(unit_valuation_difference_curr, self.product_uom_id)
             out_qty_to_invoice = product_uom._compute_quantity(out_qty_to_invoice, self.product_uom_id)
-            if not float_is_zero(unit_valuation_difference_curr * out_qty_to_invoice, precision_rounding=self.currency_id.rounding):
+            if (
+                not self.currency_id.is_zero(unit_valuation_difference_curr * out_qty_to_invoice) and
+                self.product_id.valuation != 'manual_periodic'
```

Comment:
We usually base the filters on the 'real_time' value so let's make it consistent
```suggestion
                self.product_id.valuation == 'real_time'
```